### ISSUE TITLE
Python bindings API based server push resoures error

### ISSUE THREAD

**Issue by liuli9203** [tatsuhiro-t/nghttp2#199]
I tried to write a push demo based on the nghttp2 Python bindings. However, the nghttp client just failed to receive the files pushed by the server. 
Can anybody give me a hand ? Thank you so much.  
    This is my server.py script, where folder js and script server.py is deployed in the same directory. 

```
import io, ssl 
import nghttp2 
class Handler(nghttp2.BaseRequestHandler): 
        def on_headers(self): 
                self.push(path='./js/1.js', request_headers = [('content-length', '15')],  method='GET', status=200, body='foo') 
                self.send_response(status=200, headers = [('content-type', 'text/plain')], body=io.BytesIO(b'nghttp2-python FTW')) 
server = nghttp2.HTTP2Server(('127.0.0.1', 8443), Handler, ssl=None) 
server.serve_forever()
```
Firstly, I start the python server with:
```
liuli@liuli-VM:~/Public/test$ ./server.py 
```
And then, I try to communicate with the server by starting a nghttp client. However, as shown in below, the client failed to receive the file 1.js, which was pushed by  server. 
```
liuli@liuli-VM:~/Public/nghttp2$ ./src/nghttp -v http://127.0.0.1:8443/ 
[ 0.000] Connected 
[ 0.000] send SETTINGS frame <length=12, flags=0x00, stream_id=0> 
(niv=2) 
[SETTINGS_MAX_CONCURRENT_STREAMS(0x03):100] 
[SETTINGS_INITIAL_WINDOW_SIZE(0x04):65535] 
[ 0.000] send HEADERS frame <length=37, flags=0x05, stream_id=1> 
; END_STREAM | END_HEADERS 
(padlen=0) 
; Open new stream 
:method: GET 
:path: / 
:scheme: http 
:authority: 127.0.0.1:8443 
accept: */* 
accept-encoding: gzip, deflate 
user-agent: nghttp2/0.7.10-DEV 
[ 0.000] recv SETTINGS frame <length=12, flags=0x00, stream_id=0> 
(niv=2) 
[SETTINGS_MAX_CONCURRENT_STREAMS(0x03):100] 
[SETTINGS_INITIAL_WINDOW_SIZE(0x04):65535] 
[ 0.000] recv SETTINGS frame <length=0, flags=0x01, stream_id=0> 
; ACK 
(niv=0) 
[ 0.000] recv (stream_id=1) content-length: 15 
[ 0.000] [INVALID; status=PROTOCOL_ERROR] recv PUSH_PROMISE frame <length=32, flags=0x04, stream_id=1> 
; END_HEADERS 
(padlen=0, promised_stream_id=2) 
[ 0.000] send SETTINGS frame <length=0, flags=0x01, stream_id=0> 
; ACK 
(niv=0) 
[ 0.000] send RST_STREAM frame <length=4, flags=0x00, stream_id=2> 
(error_code=PROTOCOL_ERROR(0x01)) 
[ 0.000] send RST_STREAM frame <length=4, flags=0x00, stream_id=1> 
(error_code=INTERNAL_ERROR(0x02)) 
[ 0.001] send GOAWAY frame <length=8, flags=0x00, stream_id=0> 
(last_stream_id=2, error_code=NO_ERROR(0x00), opaque_data(0)=[]) 
Some requests were not processed. total=2, processed=0 
liuli@liuli-VM:~/Public/nghttp2$  
```

**Comment by tatsuhiro-t** [comment]
I fixed a bug in python code in 1316065.  Please pull the latest change.
Also you need to replace `'./js/1.js'` with `'/js/1.js'`, since HTTP requires path must start with '/' with some exceptions (but it is not the case).

**Comment by liuli9203** [comment]
It works! Thank you so much!



Li Liu (刘利),   Master. Candidate
Institute of Computer Science & Technology ( ICST ) of Peking University, Beijing 100080, P.R. China
Email:   Liuli9203@pku.edu.cn
 
From: Tatsuhiro Tsujikawa
Date: 2015-03-31 20:55
To: tatsuhiro-t/nghttp2
CC: liuli9203
Subject: Re: [nghttp2] Python bindings API based server push resoures error (#199)
I fixed a bug in python code in 1316065. Please pull the latest change.
Also you need to replace './js/1.js' with '/js/1.js', since HTTP requires path must start with '/' with some exceptions (but it is not the case).
